### PR TITLE
Proposal for `User` schema (Please discuss 🗣️ !) 

### PR DESCRIPTION
**Pull request by brianfeister**
This was originally proposed by @j4inam on Discord here:
https://discord.com/channels/1096208627192320030/1151217204126290011/1240831524384014366

I have made the following modifications from the original proposal:

1. Dropped `reviews` (I personally would like to avoid this in the long-term. Perhaps it will become necessary but I'd rather not introduce that complexity, I see it as having minimal benefit and making us too much like Yelp, etc)
1. Added `EventSlug[]` because we want events to match URLs without doing an extra DB lookup. That said, we could run into internationalization issues with this, 
1. Replaces `roles` with => `entitlements`. I don't think our use case is enough to need roles (which would be arrays that need merging, slowing page load down) ... I'm of the opinion that we only need "capabilities" since they map 1:1 and allow users to unlock different `entitlements` (capabilities) in different orders without needing an associated role.
1. Added `mutedCurators`. Had a great conversation with someone last night where I realized that the problem of going to the site and always seeing 1,000 Alcoholics Anonymous meetings you're not interested in can be easily solved by being able to "mute" curators. Also helps in the situation where one person views the events posted by another as "inappropriate"
1. Flattened `location` and dropped complex address (Street1 / 2, city, state, zip, country) because I want us to be `latitude` and `longitude` centric. I want to verify on a map (optionally with geolocation in the browser) where the user is located
1. Dropped `password` as this will be in our auth service, not our DB

The typescript type is way easier to read:

```
type ISOString = string;
type UserId = uuid;
type EventSlug = string;

type User {
 id: string;
 firstName: string;
 lastName: string;
 mobile?: string;
 locationLatitiude?: number;
 locationLongitude?: number;
 locationAddress?: string;
 karma: number;
 email: string;
 isEmailVerified: boolean; 
 dateOfBirth: ISOString;
 entitlements: 'ingestEvent' | 'postEvent' | 'editEvent' | 'reportEvent' | 'hasSubdomain' | 'hasBranding'
 isActive: boolean;
 isVerified: boolean;
 badges?: string[]; 
 about?: string;
 avatarURL: string;
 notificationSettings: UserNotificationSettings; 
 createdAt: ISOString;
 lastLoggedIn: ISOString;
 eventsRSVP?: Event[];
 eventsAttended?: Event[];
 eventsSaved?: Event[];
 favoriteCategories?: EventSlug[];
 favoriteCurators?: UserId[];
 mutedCurators?: UserId[];
 eventsHosted?: Event[];
}
```

### CODE REVIEW

File: architecture/user.schema.json [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+      "type": "array",
+      "items": {
+        "type": "string",
+        "enum": ["ingestEvent", "postEvent", "editEvent", "reportEvent", "hasSubdomain", "hasBranding"]
+      }
+    },
+    "isActive": {
+      "type": "boolean"
+    },
+    "isVerified": {
```
Comment (brianfeister) [inline architecture/user.schema.json:96]: From @j4inam 

> // Blue tick kinda verification for popular curators

File: architecture/user.schema.json [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+      "type": "boolean"
+    },
+    "badges": {
```
Comment (brianfeister) [inline architecture/user.schema.json:99]: From @j4inam 

> // Awards or trophies unlocked for achievments/activity on the platform.

File: architecture/user.schema.json [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+    "UserNotificationSettings": {
+      "type": "object",
+      "properties": {
+        "emailNotifications": {
+          "type": "boolean"
+        },
+        "smsNotifications": {
+          "type": "boolean"
+        },
+        "pushNotifications": {
```
Comment (brianfeister) [inline architecture/user.schema.json:24]: TODO: this came from AI, we don't want this

Comment (brianfeister) [reply architecture/user.schema.json:24]: ... or maybe we do? Web native push notifications? Hmmmm 🤔

File: architecture/user.schema.json [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+      "format": "date-time"
+    },
+    "uuid": {
+      "type": "string",
+      "format": "uuid"
+    },
+    "EventSlug": {
+      "type": "string"
+    },
+    "UserNotificationSettings": {
```
Comment (brianfeister) [inline architecture/user.schema.json:15]: From @j4inam 

> // How do you want to get notified for new events posted by your fav curators, get notified for mentions, for event changes, reminders, etc. Do you want emails, sms, both ? Repeated reminders ?

File: architecture/user.schema.json [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+    "notificationSettings": {
+      "$ref": "#/definitions/UserNotificationSettings"
+    },
+    "createdAt": {
+      "$ref": "#/definitions/ISOString"
+    },
+    "lastLoggedIn": {
+      "$ref": "#/definitions/ISOString"
+    },
+    "eventsRSVP": {
```
Comment (brianfeister) [inline architecture/user.schema.json:122]: From @j4inam 

> // Separate table with list of events this user RSVP'd to

File: architecture/user.schema.json [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/Event"
+      },
+      "default": []
+    },
+    "eventsAttended": {
```
Comment (brianfeister) [inline architecture/user.schema.json:129]: From @j4inam 

> // Separate table with list of events this user attended

File: architecture/user.schema.json [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/Event"
+      },
+      "default": []
+    },
+    "eventsSaved": {
```
Comment (brianfeister) [inline architecture/user.schema.json:136]: From @j4inam 

> // Separate table with list of events this user saves or wishlists ?

File: architecture/user.schema.json [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/Event"
+      },
+      "default": []
+    },
+    "favoriteCategories": {
```
Comment (brianfeister) [inline architecture/user.schema.json:143]: From @j4inam 

> // Separate table with list this user's fav categories.

File: architecture/user.schema.json [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/EventSlug"
+      },
+      "default": []
+    },
+    "favoriteCurators": {
```
Comment (brianfeister) [inline architecture/user.schema.json:150]: From @j4inam 

> // Separate table with list this user's fav curators.

File: architecture/user.schema.json [hunk 10]
```diff
@@ -0,0 +1,188 @@
...
+      "default": []
+    },
+    "mutedCurators": {
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/uuid"
+      },
+      "default": []
+    },
+    "eventsHosted": {
```
Comment (brianfeister) [inline architecture/user.schema.json:164]: From @j4inam 

> // Separate table with list of events this user hosted.

**Comment by j4inam** [comment]
Moved to #62

**Pull request closed without merging**